### PR TITLE
Refactor `exp:search:total_results`

### LINKED ISSUE #1193: Parse {exp:search:total_results} as any other template tag
**Is your feature request related to a problem? Please describe.**
Currently. despite `{exp:search:total_results}` can be used as a variable pair, it has no difference from a single variable

**Describe the solution you'd like**

I would like to be able to code something like:

```
{exp:search:total_results}
    <p>
        {total_results} found.
    </p>
    {if no_results}
        <p>No results found.</p>
    {/if}
{/exp:search:total_results}
```

- [x] I am capable and would like to work on implementation of this feature if it is considered.

### PR DESCRIPTION
**Pull request by robsonsobral**
## Overview

Refactor `exp:search:total_results` to be parsed as any template, allowing `no_results` and other conditionals.

Resolves [#1193](https://github.com/ExpressionEngine/ExpressionEngine/issues/1193).

## Nature of This Change

- [ ] 🐛 Fixes a bug
- [x] 🚀 Implements a new feature
- [x] 🛁 Refactors existing code
- [ ] 💅 Fixes coding style
- [ ] ✅ Adds tests
- [ ] 👽 Adds new dependency
- [ ] 🔥 Removes unused files / code
- [ ] 🔒 Improves security 

## Is this backwards compatible?

- [x] Yes
- [ ] No

## Documentation

User Guide Pull Request: https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/NNN

### CODE REVIEW

**Comment by robsonsobral** [comment]
I'm not sure about the third parameter on `ee()->TMPL->parse_variables_row(ee()->TMPL->tagdata, (array) $query->row(), true);`.

File: system/ee/ExpressionEngine/Addons/search/mod.search.php
Comment (intoeetive) [inline system/ee/ExpressionEngine/Addons/search/mod.search.php]: No need to do typecasting for `(array) $query->row()`, you can just use `$query->row_array()`. Also since 3rd parameter is default, it can just be skipped (which is what we usually do)

**Review by intoeetive — CHANGES_REQUESTED** [review]
Requested small change.

I think we also need to have this added to Cypress test to make sure this does not break existing code where people has like `{if total_results = 0}`

**Comment by robsonsobral** [comment]
> Requested small change.
> 
> I think we also need to have this added to Cypress test to make sure this does not break existing code where people has like `{if total_results = 0}`

@intoeetive , I don't think anybody ever used that, because it never worked. This is the reason I proposed this change.

**Comment by robsonsobral** [comment]
Despite I've added the tests in the dark, I guess the fail is caused by outdated branch. I gonna check it.

**Comment by intoeetive** [comment]
It looks like this PR is assuming that https://github.com/ExpressionEngine/ExpressionEngine/pull/1674 is already merged. However it is not. Maybe worth combining the two into a single PR?

**Comment by robsonsobral** [comment]
I let some tests commented. They can be enabled after the merge of #1674

**Comment by intoeetive** [comment]
I'm not sure commenting out part of the test is the best way to make it pass =)

Can we just make the "no results" test use separate "no results" page?

**Comment by robsonsobral** [comment]
> I'm not sure commenting out part of the test is the best way to make it pass =)



Oh, I'm sorry. My intention was to not to "make them pass", but to save us from write them back after the merge. They only make sense after the merge. To write 2 forms and 2 pages to test only one feature which will work the same in both situations, looks a little too much.

But... In case you tink it's best, I can do it!

**Comment by intoeetive** [comment]
If you think it's worth just combining these two PRs into one, just go ahead and do it. You have my approval on both ones, so it's just the matter of tests

**Comment by robsonsobral** [comment]
DONE!

**Comment by robsonsobral** [comment]
For reviewers to keep an eye on:

- instructions on how to fill the changelog files properly. I'm always making a mess on them;

**Review by bryannielsen — DISMISSED** [review]

**Comment by robsonsobral** [comment]
Do you want me to fix this conflicts, @intoeetive ?

**Review by intoeetive — CHANGES_REQUESTED** [review]
@robsonsobral yes please, if you could resolve the conflicts that would be great (and sorry for being slow here)

**Pull request closed without merging**
